syntax = "proto2";

package amarlon;

message SpellData {
	optional int32  id          = 1;
	optional int32  level       = 2;
	optional int32  char_class  = 3;
	optional int32  target_type = 4 [default=0];
	optional string name        = 5;
	optional string description = 6;
	optional bool   passive     = 7 [default=false];
	optional int32  radius      = 8 [default=0];
	optional int32  range       = 9 [default=0];
}

message SpellsData {
	repeated SpellData spell = 1;
}

message SpellBookData {
	message SlotData{
		optional int32 level     = 1;
		optional bool isPrepared = 2 [default=false];
		optional int32 spell_id  = 3;
	}

	repeated int32    knownSpells = 1;	
	repeated SlotData spellSlots  = 2;	
}